### PR TITLE
Pendle protocol

### PR DESCRIPTION
**Pull request by zhufengthehemera**

### CODE REVIEW

File: indexer/modules/custom/pendle/export_pendle_pools_job.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        pools = {}
+        for log in logs:
+            if log.address not in [
+                self.user_defined_config["pendle_market_factory_address"],
+                self.user_defined_config["pendle_market_factory_v3_address"],
+            ]:
+                continue
+            if log.topic0 not in [create_market_event.get_signature(), create_market_event_v3.get_signature()]:
+                continue
+            market_address = Web3.to_checksum_address("0x" + log.topic1[26:])
```
Comment (shanshuo0918) [inline indexer/modules/custom/pendle/export_pendle_pools_job.py:71]: We should have a util function for this

File: indexer/modules/custom/pendle/export_pendle_balance_job.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                target=market_address,
+                function_abi=market_active_balance,
+                parameters=[user_address],
+                block_number=block_number,
+            )
+            call_to_domain_data[id(c)] = domain
+            calls.append(c)
+
+            # market totalActiveSupply
+            c = Call(
```
Comment (shanshuo0918) [inline indexer/modules/custom/pendle/export_pendle_balance_job.py:123]: I think you don't need to call this for every user?

**Review by shanshuo0918 — APPROVED** [review]

File: indexer/modules/custom/pendle/export_pendle_pools_job.py (hunk 1 above)
Comment (zhufengthehemera) [reply indexer/modules/custom/pendle/export_pendle_pools_job.py:71]: t has already been implemented in the Lido pr. The code will be updated after the merge.

File: indexer/modules/custom/pendle/export_pendle_balance_job.py (hunk 2 above)
Comment (zhufengthehemera) [reply indexer/modules/custom/pendle/export_pendle_balance_job.py:124]: Each time a user's activityBalance changes, the totalActiveSupply will also be updated, so it needs to be requested. Perhaps the totalActiveSupply field can be separated into its own independent table.

**Pull request merged**